ui/pages/users/index.js

```diff
@@ -343,9 +343,11 @@ export default function Users() {
                 }
               )
 
-              // can only delete users that exist within Infra which are not the currently logged in user
+              // can only delete Infra users where password authn is the only authn method
               if (
-                info.row.original.id === user?.id ||
+                (info.row.original.id === user?.id &&
+                  info.row.original.providerNames?.filter(n => n != 'infra')
```

Comment:
I think you should still be able to delete Infra users when they have logged in via other sources so this restriction on `infra` being the only provider isn't quite right